cmd/zpool/zpool_main.c

```diff
@@ -2471,61 +2474,587 @@ zpool_do_import(int argc, char **argv)
+		val = vs->vs_queue_active[list[i]];
+		print_one_stat(val, format, column_width, cb->cb_scripted);
+	}
+}
+
+static void
+print_iostat_latency(vdev_stat_t *vs, iostat_cbdata_t *cb)
+{
+	unsigned int column_width = default_column_width(cb, VS_LATENCY);
+	enum zfs_nicenum_format format;
+
+	if (cb->cb_literal)
+		format = ZFS_NICENUM_RAW;
+	else
+		format = ZFS_NICENUM_TIME;
+
+	print_one_stat(single_histo_average(vs->vs_total_histo[ZIO_TYPE_READ],
+		VDEV_HISTO_BUCKETS), format, column_width, cb->cb_scripted);
+	print_one_stat(single_histo_average(vs->vs_total_histo[ZIO_TYPE_WRITE],
+		VDEV_HISTO_BUCKETS), format, column_width, cb->cb_scripted);
+	print_one_stat(single_histo_average(vs->vs_disk_histo[ZIO_TYPE_READ],
+		VDEV_HISTO_BUCKETS), format, column_width, cb->cb_scripted);
+	print_one_stat(single_histo_average(vs->vs_disk_histo[ZIO_TYPE_WRITE],
+		VDEV_HISTO_BUCKETS), format, column_width, cb->cb_scripted);
+}
+
+/*
+ * Print default statistics (capacity/operations/bandwidth)
+ */
+static void
+print_iostat_default(vdev_stat_t *vs, iostat_cbdata_t *cb)
+{
+	unsigned int column_width = default_column_width(cb, VS_DEFAULT);
+	enum zfs_nicenum_format format;
+	char na;	/* char to print for "not applicable" values */
+
+	if (cb->cb_literal) {
+		format = ZFS_NICENUM_RAW;
+		na = '0';
+	} else {
+		format = ZFS_NICENUM_1024;
+		na = '-';
+	}
+
+	/* only toplevel vdevs have capacity stats */
+	if (vs->vs_space == 0) {
+		if (cb->cb_scripted)
+			printf("\t%c\t\%c", na, na);
+		else
+			printf("  %*c  %*c", column_width, na, column_width,
+				na);
+	} else {
+		print_one_stat(vs->vs_alloc, format, column_width,
+							cb->cb_scripted);
+		print_one_stat(vs->vs_space - vs->vs_alloc, format,
+						column_width, cb->cb_scripted);
+	}
+
+	print_one_stat((uint64_t)(vs->vs_ops[ZIO_TYPE_READ]),
+			format, column_width, cb->cb_scripted);
+	print_one_stat((uint64_t)(vs->vs_ops[ZIO_TYPE_WRITE]),
+			format, column_width, cb->cb_scripted);
+	print_one_stat((uint64_t)(vs->vs_bytes[ZIO_TYPE_READ]),
+			format, column_width, cb->cb_scripted);
+	print_one_stat((uint64_t)(vs->vs_bytes[ZIO_TYPE_WRITE]),
+			format, column_width, cb->cb_scripted);
 }
 
 /*
  * Print out all the statistics for the given vdev.  This can either be the
  * toplevel configuration, or called recursively.  If 'name' is NULL, then this
  * is a verbose output, and we don't want to display the toplevel pool stats.
+ *
+ * Returns the number of stat lines printed.
  */
-void
+unsigned int
 print_vdev_stats(zpool_handle_t *zhp, const char *name, nvlist_t *oldnv,
     nvlist_t *newnv, iostat_cbdata_t *cb, int depth)
 {
 	nvlist_t **oldchild, **newchild;
 	uint_t c, children;
-	vdev_stat_t *oldvs, *newvs;
+	vdev_stat_t *oldvs, *newvs, *calcvs;
 	vdev_stat_t zerovs = { 0 };
-	uint64_t tdelta;
-	double scale;
 	char *vname;
+	int i;
+	int ret = 0;
+
+	calcvs = safe_malloc(sizeof (*calcvs));
+	ASSERT(calcvs);
+	if (!calcvs) {
+		printf("Not enough memory to print\n");
+		return (ret);
+	}
```

Comment:
You can safely drop the `!calcvs` check here.  `safe_malloc()` performs it internally and will log and error and exit unconditionally if the allocation fails.